/*!
* @file
* @brief IKeyFactory
*
* @copyright Copyright (C) University of Bristol 2017
*    This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. 
*    If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
*    See LICENSE file for details.
* @date 02/02/2018
* @author Richard Collins <richard.collins@bristol.ac.uk>
*/

syntax = "proto3";
import public "google/protobuf/empty.proto";
import public "Messages.proto";

package cqp.remote;

/**
 * @brief IKey Interface
 * Interface for requesting shared keys, this should only be exposed to
 * clients within the local site
 */
service IKey
{

    /**
    * Query the key provider for a list of other sites which it is exchanging key with
    * @param response The destinations this site provide keys for
    * @return status
    */
    rpc GetKeyStores(google.protobuf.Empty) returns (SiteList);

    /**
    * @brief GetNewKey
    * Get a shared key which is valid between two sites
    * @param request The key type to get, to get a new key, leave the keyid unassigned
    * @param response The keyid and it's value
    * @return status
    */
    rpc GetSharedKey(KeyRequest) returns (SharedKey);

}
